prow/flagutil/github.go

```diff
@@ -61,11 +63,19 @@ func (o *GitHubOptions) addFlags(wantDefaultGitHubTokenPath bool, fs *flag.FlagS
 // Validate validates GitHub options.
 func (o *GitHubOptions) Validate(dryRun bool) error {
 	for _, uri := range o.endpoint.Strings() {
-		if _, err := url.ParseRequestURI(uri); err != nil {
+		if uri == "" {
+			uri = github.DefaultAPIEndpoint
```

Comment:
The default is already specified above. If `uri == ""` here then `--github-endpoint` was explicitly set to the empty value which should be an error.

Reply:
I am using the default in validate also, as some testing skips addFlags step, so this value would be empty. Is it possible to have a follow-up PR for this, I have had a bit of back and forth on this requirement from Steve and Fejta. 

The general consensus was to not couple validation logic across individual components.